### PR TITLE
feat: Verify the rayo JvbRoomName header if it exists.

### PR DESCRIPTION
**Pull request by bgrozev**
Note it's just a precaution, prosody already verifies it.

### CODE REVIEW

File: jicofo/src/main/kotlin/org/jitsi/jicofo/xmpp/JigasiIqHandler.kt [hunk 1]
```diff
@@ -82,6 +78,17 @@ class JigasiIqHandler(
             }
         }
 
+        val roomNameHeader = request.iq.getHeader("JvbRoomName")
```
Comment (damencho) [inline jicofo/src/main/kotlin/org/jitsi/jicofo/xmpp/JigasiIqHandler.kt:81]: Does this search the whole structure of the stanza? Cause the dial-iq is like:
```
<iq to = 'focus@auth.meet.jitsi/focus' type = 'set' xml:lang = 'en'
    id = ''
    from = 'damencho@conference.tenant.v1.meet.jitsi/36ff70d6'>
    <dial to = 'jitsi_meet_transcribe' xmlns = 'urn:xmpp:rayo:1' from = 'fromnumber'>
        <header value = 'damencho@conference.tenant.v1.meet.jitsi' name = 'JvbRoomName' />
    </dial>
</iq>
```

Comment (bgrozev) [reply jicofo/src/main/kotlin/org/jitsi/jicofo/xmpp/JigasiIqHandler.kt:81]: I think so, this is the `getHeader` from `RayoIq`
https://github.com/jitsi/jitsi-xmpp-extensions/blob/3816e5a154160056290c2a2fbff1ab63cbac95e0/src/main/java/org/jitsi/xmpp/extensions/rayo/RayoIq.java#L55

Comment (damencho) [reply jicofo/src/main/kotlin/org/jitsi/jicofo/xmpp/JigasiIqHandler.kt:81]: But that gets all direct child extensions and the only child of the IQ is a dial extension. The header is the child of the dial extension.

Comment (damencho) [reply jicofo/src/main/kotlin/org/jitsi/jicofo/xmpp/JigasiIqHandler.kt:81]: Oh this method gets the DialIq `request: IqRequest<DialIq>`. Sorry for the noise.

**Review by damencho — DISMISSED** [review]

File: jicofo/src/main/kotlin/org/jitsi/jicofo/xmpp/JigasiIqHandler.kt (hunk 1 above)
Comment (bgrozev) [reply jicofo/src/main/kotlin/org/jitsi/jicofo/xmpp/JigasiIqHandler.kt:81]: I think this test verifies it works as expected?
https://github.com/jitsi/jitsi-xmpp-extensions/blob/3816e5a154160056290c2a2fbff1ab63cbac95e0/src/test/java/org/jitsi/xmpp/extensions/rayo/DialIqProviderTest.java#L138

**Review by damencho — APPROVED** [review]

**Pull request merged**